src/silo/common/fasta_reader.cpp

```diff
@@ -0,0 +1,43 @@
+#include "silo/common/fasta_reader.h"
+
+silo::FastaReader::FastaReader(std::string in_file_name)
+    : in_file(in_file_name) {}
+
+bool silo::FastaReader::nextKey(std::string& key) {
+   std::string key_with_prefix;
+   if (!getline(in_file.getInputStream(), key_with_prefix)) {
+      return false;
+   }
+
+   if (key_with_prefix.at(0) != '>') {
+      return false;
```

Comment:
I think we should throw exceptions instead.

Can't we simply return the key instead of populating a passed reference?

Reply:
I initially had it like that, but this would copy too much data for the genomes

Reply:
Exception is a good idea, what would you suggest, since this is now no longer in `silo/preprocesssing` but in `silo/common`? How about `silo/common/FastaFormatException`

Reply:
> I initially had it like that, but this would copy too much data for the genomes

And I wanted to have the key/genome consistent